### ISSUE TITLE
Only X axis homing switch works for all 3 axis

### ISSUE THREAD

**Issue by karoria** [synthetos/g2#393]
Hi friends, I am using g2 core (master) with arduino due. All the 3 axis works perfectly except the fact that only one output pin on due (pin 14) acts as homing switch for X, Y and Z axis. I have flashed the board with all NO and all NC settings but the result remains same. Kindly help. One more question is, can I do homing X and Y simultaneously after Z is homed? Please help...

**Comment by justinclift** [comment]
When you say "X and Y simultaneously", are you meaning to start the homing with a single command, and it's ok for one axis to home first, then the other to then home?

Or are you meaning for both the X and Y axes to be started with a single command, and the motors for both to run at the same time?

From memory - I haven't tested this in a few months as I don't have my equipment with me - you can start the homing for multiple axes with one command.  But each axis' motor(s) will do the homing, one axis at a time.

With the one output pin, I'm not sure what's causing the problem you're having.  When I was doing stuff with g2 a few months ago (on a Shapeoko 3), I had homing switches on X, Y, and Z.  g2 worked ok with 3 homing switches then.

At a guess, the problem you're having is **probably** something to do with the config file you're using.  Are you using the defaults, or something else?

**Comment by karoria** [comment]
Thanks for the prompt response. Ok. Homing all the axis one by one is ok for me if there is no evident solution.
Now coming to the switches, I use the default gshield configuration with only tweaks for motors microstepping and relevant data. I use the connection as given in following link:
https://github.com/synthetos/g2/wiki/Arduino-DUE-Pinout-for-g2core
My problem is, while homing by g28.2x0y0z0 the z, x and y don't home as they should. I mean z and y limit switches don't work with default wiring as per diagram. For all the axis, Xmin switch works as homing switch. This i found by manually firing the switch while y and z were homing as their respective switches didn't work!
In other words, if i use NC switches for homing each axis and i connect them in series and connect it to pin 14 which is for Xmin while other end connected to ground, my machine homes accurately. But that is not i want as we have dedicated homing connections for each axis. Is it somewhere in settings we can isolate these axis homing switch inputs and I am just missing that point?

**Comment by justinclift** [comment]
Ouch.  It'll (probably, bugs aside) be either a wiring problem, or something might not be totally correct with your config file.

I don't personally have my equipment with me any more :frowning:, so I there's a pretty hard limit to the bits I can test in order to investigate.

What **might** help though, is this g2core config file I was using (successfully) with my Shapeoko 3 converted to an Arduino Due + g2core:

&nbsp; &nbsp; https://github.com/justinclift/g2/blob/100.26-shapeoko3/g2core/settings/settings_shapeoko3.h

That was also for an Arduino Due hooked up using that pinout guide, so (I think) the config file should match.  With the exception of the X/Y/Z axis travel limits, depending on the size of your machine.

As a thought, with that config file you might as well just compile that exact branch of g2core I was using as it's based on the stable master branch, and plays nicely with CNCjs, UGC, etc:

&nbsp; &nbsp; https://github.com/justinclift/g2/commits/100.26-shapeoko3

Hopefully that gets you up and running.  Or at least closer. :smile:

**Comment by karoria** [comment]
Thanks a lot @justinclift for the links. I will give it a try today only and will share the results here :)

**Comment by karoria** [comment]
Hurraaay...Solved!!! Thanks a lot again. My problem was, I had set xhi (x homing input), yhi and zhi all to 1, as I was in impression that 1 is to enable and 0 is to disable!! After studying your settings file, I got that it was for physical pin numbers.
Now everything works well. BTW, I am using my version of firmware only, with above corrections made.
Thanks again for being so active here and guiding us...

**Comment by justinclift** [comment]
Cool!  That's really good news @karoria.  And yep, you're welcome. :smile:
